source/administration/incident-response-application.rst

```diff
@@ -0,0 +1,466 @@
+          }
+        ]
+      }
+    ]
+
+There are four components to an incident app workflow schema: triggers, steps, actions and transitions. Each one is explained in more detail below.
+
+Trigger
+^^^^^^^^^^^^^^^
+
+A trigger is a condition on which a workflow can be started.
+
+.. csv-table::
+    :header: "Field", "Description", "Type", "Required"
+
+    "name", "The name of the trigger", "string", "Yes. This field must be unique."
+    "type", "The type of trigger. This refers to the internal name of the trigger being configured", "string, "Yes"
+    "*additional fields", "Additional fields based on the type of trigger. See below for more details.", "string", "Yes"
+
+Text Trigger (type: ``text``)
+*******************************
+
+An incident can be created when a text trigger is seen. The specified message then becomes the description of the incident.
+
+For example, you can specify "#s1critical" as the text trigger and any post that contains this keyword is treated as an incident and starts the workflow.
+
+.. csv-table::
+    :header: "Field", "Description", "Type", "Required"
+
+    "channel", "The channel(s) to look for the trigger. If none specified, all channels will be watched.", "[]string", "No"
+    "team", "The team(s) to watch for the trigger in. If none specified, all teams will be watched.", "[]string, "No"
+    "match", "The text string to match on", "string", "If ``match_regex`` not set, yes"
+    "match_regex", "The regex to match on", "string", "If ``match`` not set, yes"
+
+// TODO Verify JSON schema and add an example here.
+
+Step
+^^^^^^^^^^^^^^^
+
+A step is the representation of a state the app workflow can be in. For now it contains a set of actions to be performed when it is started and when it is finished.
+
+.. csv-table::
+    :header: "Field", "Description", "Type", "Required"
+
+    "name", "The name of the step", "string", "Yes. This field must be unique."
+    "start_actions", "Actions to perform when the step is reached.", "[]Action", "If ``finished_actions`` not set, yes"
+    "finished_actions", "Actions to perform when the step is finished.", "[]Action", "If ``start_actions`` not set, yes"
+
+Action
+^^^^^^^^^^^^^^^
+
+Actions are performed when steps are started and finished.
+
+.. csv-table::
+    :header: "Field", "Description", "Type", "Required"
+
+    "name", "The name of the action", "string", "Yes. This field must be unique."
+    "type", "The type of action to perform", "string", "Yes"
+    "*additional fields", "Additional fields based on the type of action. See below for more details.", "string", "Yes"
+
+Create Channel (type: ``create_channel``)
+*******************************************
+
+The create channel action creates a channel with the given parameters. The users listed will be automatically invited.
+
+.. csv-table::
+    :header: "Field", "Description", "Type", "Required"
+
+    "channel", "A channel object supporting fields from model.Channel", "model.Channel", "Yes"
```

Comment:
Should switch out for specific supported fields. See: https://github.com/mattermost/mattermost-plugin-workflow/blob/master/server/action/createchannel.go#L14